python/ml4ir/model/metrics/metric_factory.py

```diff
@@ -1,16 +1,28 @@
 from ml4ir.config.keys import MetricKey
-
-# import tensorflow as tf
+from ml4ir.model.metrics.metrics_impl import MRR, ACR, MeanMetricWrapper
+from typing import Union, Type
+from tensorflow.keras.metrics import Metric
 
 
 def get_metric(metric_key):
     if metric_key == MetricKey.MRR:
-        raise NotImplementedError
+        return MRR
     elif metric_key == MetricKey.ACR:
-        raise NotImplementedError
+        return ACR
     elif metric_key == MetricKey.NDCG:
         raise NotImplementedError
     elif metric_key == MetricKey.CATEGORICAL_ACCURACY:
         return "categorical_accuracy"
     else:
         raise NotImplementedError
+
+
+def get_metric_impl(metric: Union[str, Type[MeanMetricWrapper]], **kwargs) -> Union[str, Metric]:
+    if isinstance(metric, str):
```

Comment:
when do we use this if branch?

Reply:
For inbuilt tf metrics - https://www.tensorflow.org/api_docs/python/tf/keras/metrics/CategoricalAccuracy

Look at this above
```
elif metric_key == MetricKey.CATEGORICAL_ACCURACY:
        return "categorical_accuracy"	        return "categorical_accuracy"
```

Reply:
isn't there an actual function in the core tf library we can refer to, instead of by name?  Like using a wrapper around `tf.keras.metrics.categorical_accuracy`?

Reply:
Yes, I was hoping you wouldn't catch it. Let me change it.

Reply:
whenever you see your method signatures have super weird `Union` types (like `[str,Type[MeanMetricWrapper]]`), there's a hint that maybe it could be refactored.